AsyncDisplayKit/Private/ASEnvironmentInternal.mm

```diff
@@ -115,33 +115,33 @@ ASEnvironmentState ASEnvironmentMergeObjectAndState(ASEnvironmentState environme
     
     // For every field check if the parent value is equal to the default than propegate up the child value to
     // the parent
-    if (parentState.spacingBefore != defaultState.spacingBefore) {
+    if (parentState.spacingBefore == defaultState.spacingBefore) {
       parentState.spacingBefore = state.spacingBefore;
     }
-    if (parentState.spacingAfter != defaultState.spacingAfter) {
+    if (parentState.spacingAfter == defaultState.spacingAfter) {
       parentState.spacingAfter = state.spacingAfter;
     }
-    if (parentState.alignSelf != defaultState.alignSelf) {
-      parentState.alignSelf = defaultState.alignSelf;
+    if (parentState.alignSelf == defaultState.alignSelf) {
+      parentState.alignSelf = state.alignSelf;
     }
-    if (parentState.flexGrow != defaultState.flexGrow) {
-      parentState.flexGrow = defaultState.flexGrow;
+    if (parentState.flexGrow == defaultState.flexGrow) {
```

Comment:
@maicki just wanted to check with you - on all the changes in this PR, but these do seem like important ones / look correct to me.